op/model.py

```diff
@@ -277,7 +277,14 @@ def set_spec(self, spec, k8s_resources=None):
 
 
 class ModelError(Exception):
-    pass
+
+    @classmethod
+    def from_exception(cls, exception):
+        if isinstance(exception, CalledProcessError):
+            if exception.cmd[0] in ('relation-list', 'relation-get', 'relation-set') and exception.returncode == 2\
```

Comment:
ModelError is the general piece. It doesn't make sense for it to know about something extremely specific such as the exact error message that a given tool for a given implementation will return. It doesn't even make sense for it to know about CalledProcessError.. it should be the other way around: whoever creates the error needs to know how to deal with it.